### DISCUSSION TITLE
GetRecords<dynamic> Not escaping quotes while GetField() seems to do so?

### DISCUSSION THREAD

**Discussion by Quackers2446** [JoshClose/CsvHelper#1845] [Q&A] [answered]
Hi there!

I'm trying to read in a CSV file using GetRecords<dynamic>, but it doesn't seem to escape quotes while GetField() seems to do so.

Here's what my CSV looks like:

```
name,id
"""Hello"", World", 123
Bob,678
```

Here's the code I'm using to read by hand:

```
var records = new List<Foo>();
reader.Read();
reader.ReadHeader();

while (reader.Read())
{
	var record = new Foo
	{
		name = reader.GetField("name"),
		id = reader.GetField("id")
	};
	records.Add(record);
}
```

When reading through my CSV using GetRecords<dynamic>, I get `"Hello", World` as the first field, however, when reading by hand I get `"\"Hello\", World"` instead.

Currently the API I'm using doesn't like `"Hello", World`, but at the same time reading by hand is not an option. Is there any way to escape quotes using GetRecords? I'd like to get `"\"Hello\", World"` as my output if possible!

Thanks!

**Comment by JoshClose** [comment, marked as answer]
I'm not getting that result when reading by hand.

```cs
void Main()
{
    var s = new StringBuilder();
    s.Append("name,id\r\n");
    s.Append("\"\"\"Hello\"\", World\", 123\r\n");
    s.Append("Bob,678\r\n");
    var records = new List<Foo>();
    var config = new CsvConfiguration(CultureInfo.InvariantCulture)
    {
    };
    using (var reader = new StringReader(s.ToString()))
    using (var csv = new CsvReader(reader, config))
    {
        csv.Read();
        csv.ReadHeader();
        while (csv.Read())
        {
            var record = new Foo
            {          
                Id = csv.GetField<int>("id"),
                Name = csv.GetField("name"),
            };
            records.Add(record);
        }
        
        records.Dump();
    }
}

private class Foo
{
    public int Id { get; set; }
    public string Name { get; set; }
}
```
![image](https://user-images.githubusercontent.com/114971/129633148-4d72bfd6-b0a7-431b-8e67-16a57fd97028.png)

I could have to crate a way to get the raw field in the parser to do what you want to do.

**Reply by Quackers2446** [reply]
Huh, that's interesting.. So there's no current functionality to have it do that? Sorry, I'm very new to using the Csv helper.

If you could look into that that would be amazing!

**Reply by JoshClose** [reply]
No there isn't. The parser strips out the escapes. This would be pretty simple to do though.

**Reply by Quackers2446** [reply]
I see, thank you for the fast response!! I managed to find the correct place in my code for a Replace() to insert an escape and it's all working now! Thank you for your help!!
